File: 121. Best Time to Buy and Sell Stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minprice = 2**31 - 1
        maxprofit = 0
        for i in prices:
            minprice = min(minprice, i)
            curr = i - minprice
            maxprofit = max(maxprofit, curr)
        return maxprofit

File: 121. Best Time to Buy and Sell Stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int currmax=0, totalmax=0;
        for (int i=1; i<prices.length; i++) {
            currmax = Math.max(currmax+prices[i]-prices[i-1], prices[i]-prices[i-1]);
            totalmax = Math.max(totalmax, currmax);
        }
        return totalmax;
    }
}